statsmodels/discrete/tests/test_constrained.py

```diff
@@ -65,61 +65,54 @@ def test_basic(self):
 
 
     def test_basic_method(self):
-        if hasattr(self, 'res1m'):
-            res1 = (self.res1m if not hasattr(self.res1m, '_results')
-                               else self.res1m._results)
-            res2 = self.res2
-            assert_allclose(res1.params, res2.params[self.idx], rtol=1e-6)
-
-            # when a parameter is fixed, the Stata has bse=nan, we have bse=0
-            mask = (res1.bse == 0) & np.isnan(res2.bse[self.idx])
-            assert_allclose(res1.bse[~mask], res2.bse[self.idx][~mask], rtol=1e-6)
-
-            tvalues = res2.params_table[self.idx, 2]
-            # when a parameter is fixed, the Stata has tvalue=nan, we have tvalue=inf
-            mask = np.isinf(res1.tvalues) & np.isnan(tvalues)
-            assert_allclose(res1.tvalues[~mask], tvalues[~mask], rtol=1e-6)
-            pvalues = res2.params_table[self.idx, 3]
-            # note most pvalues are very small
-            # examples so far agree at 8 or more decimal, but rtol is stricter
-            # see above
-            mask = (res1.pvalues == 0) & np.isnan(pvalues)
-            assert_allclose(res1.pvalues[~mask], pvalues[~mask], rtol=5e-5)
-
-            ci_low = res2.params_table[self.idx, 4]
-            ci_upp = res2.params_table[self.idx, 5]
-            ci = np.column_stack((ci_low, ci_upp))
-            # note most pvalues are very small
-            # examples so far agree at 8 or more decimal, but rtol is stricter
-            # see above: nan versus value
-            assert_allclose(res1.conf_int()[~np.isnan(ci)], ci[~np.isnan(ci)], rtol=5e-5)
-
-            #other
-            assert_allclose(res1.llf, res2.ll, rtol=1e-6)
-            assert_equal(res1.df_model, res2.df_m)
-            # Stata doesn't have df_resid
-            df_r = res2.N - res2.df_m - 1
-            assert_equal(res1.df_resid, df_r)
-        else:
-            pytest.skip("not available yet")
-
-    def test_other(self):
-        # some results may not be valid or available for all models
-        if hasattr(self, 'res1m'):
-            res1 = self.res1m
-            res2 = self.res2
-
-            if hasattr(res2, 'll_0'):
-                assert_allclose(res1.llnull, res2.ll_0, rtol=1e-6)
-            else:
-                if DEBUG:
-                    import warnings
-                    message = ('test: ll_0 not available, llnull=%6.4F'
-                                                                % res1.llnull)
-                    warnings.warn(message)
+        res1 = (self.res1m if not hasattr(self.res1m, '_results')
```

Comment:
You could fix the indentation.  If not hasattr(): skip then no indentation needed below.

Reply:
josef and I are working together pretty well in this thread, I'd like to keep that going by not pushing on this.

Reply:
What's the problem with intentation?
this looks like a regular if else to me.

Reply:
The function is

```python
if cond:
    Lots of lines fo test code:
else:
    skip
```

Better hygiene is to do

```python
if not cond:
    skip
lots of test code (<- 1 level less indent, and no need to read to the bottom to figure out what is going on)
```

This is just hygiene, so clearly not essential.